examples/daal/cpp/source/utils/service.h

```diff
@@ -658,50 +658,55 @@ void printNumericTables(daal::data_management::NumericTablePtr dataTable1,
                                      interval);
 }
 
-bool checkFileIsAvailable(std::string filename, bool needExit = false) {
-    std::ifstream file(filename.c_str());
-    if (file.good()) {
-        return true;
-    }
-    else {
-        std::cout << "Can't open file " << filename << std::endl;
-        if (needExit) {
-            exit(fileError);
+// The function tries to find the file `name` in several possible directories.
+// This is useful because CMake and Bazel may run the program from different working directories,
+// so relative paths to data files can differ.
+inline std::string get_data_path(const std::string &name) {
+    const std::vector<std::string> paths = { "../data", "examples/daal/data" };
+
+    for (const auto &path : paths) {
+        std::string try_path = path + "/" + name;
+        if (std::ifstream{ try_path }.good()) {
+            return try_path;
         }
-        return false;
     }
+
+    return name;
 }
 
 void checkArguments(int argc, char *argv[], int count, ...) {
     std::string **filelist = new std::string *[count];
+
```

Comment:
`filelist` have to be defined as:
`const std::string ** filelist = new const std::string *[count];` in order to retain constness.

Reply:
in this case its not possible to do like
`*(filelist[i]) = get_data_path(*(filelist[i]));`